signac/contrib/collection.py

```diff
@@ -936,9 +936,9 @@ def _find(self, filter=None, limit=0):
             if not _valid_filter(filter):
                 raise ValueError(filter)
             result = self._find_result(filter)
-            return set(islice(result, limit if limit else None))
         else:
-            return set(islice(self._docs.keys(), limit if limit else None))
+            result = self._docs.keys()
+        return set(islice(result, limit if limit else None))
```

Comment:
Unrelated change?

Reply:
I pushed this small change directly to `next` because it makes code more readable. It's covered by tests.